docs/content/docs/install/_index.md

```diff
@@ -19,18 +19,32 @@ Unpack the `osm` binary and add it to `$PATH` to get started.
 
 ### From Source (Linux, MacOS)
 
-Building OSM from source requires more steps but is the best way to test the latest changes and useful in a development environment.
+Building OSM CLI from source requires more steps but is the best way to test the latest changes and useful in a development environment.
 
 You must have a working [Go](https://golang.org/doc/install) environment.
 
 ```console
 $ git clone git@github.com:openservicemesh/osm.git
 $ cd osm
+```
+
+When OSM CLI is built from source, it might not be compatible with the default control plane images derived from the latest released version. In such cases, use the `latest_main` image tag to specify that the CLI should use the latest control plane images built from the `main` branch. It is recommended to build the OSM CLI binary from the latest commit on the `main` branch while using the `latest_main` image tag for compatibility reasons.
```

Comment:
Should we make the "default" recommendation when building the CLI from source to also build the images from the same source? That's guaranteed to work, or at least if issues are found then those are definitely bugs that need to be fixed. Then the `latest_main` images can be a convenient shortcut that will _usually_ work, but there's still some incompatibility risks.

Reply:
I thought about this. I think building controller images from source also means we would need to document information regarding container registry, tags, ImagePullSecrets etc. In the short term, `latest_main` could provide a reasonable alternative.